### DISCUSSION TITLE
Invalid byte shown from the previous instruction

### DISCUSSION THREAD

**Discussion by ciapatakoza** [Vector35/binaryninja-api#4506] [Architectures] [answered]
Hi,
I am currently working with the Xtensa architecture plugin, I have just noticed some weird bug. For some reason Binary Ninja put "??" in some places. I have inspected that, and it seems that it actually show an byte from the previous instruction for some reason. The example is shown on the screenshot. As you can see the 0x28abe instruction has 8B, so it should end at 0x28ac5. But, we see in the next line 0x28ac1 null byte marked as "??". I understand that this is probably some bug with the plugin. But could you give any pointer what could go wrong, why it show and address below an instruction that should contain it? As the size of the previous instruction was set to 8 for sure.

![image](https://github.com/Vector35/binaryninja-api/assets/49192325/7d05e552-7017-4d21-b753-a32cb90b99da)

**Comment by xusheng6** [comment]
Both `get_instruction_info` and `get_instruction_text` need to return the instruction length. Do they both return 8 for the instruction at 0x28abe?

**Comment by ciapatakoza** [comment, marked as answer]
Ok, nvm. I have broken control flow while calculating FalseBranch address.
